### PR TITLE
Update view-transitions.mdx

### PR DESCRIPTION
**Pull request by Dillonpw**
Potential resolution to issue #8787 with an extended explanation of custom View Transitions

#### Description (required)

first iteration of an update to the View Transitions documentation, going into more detail on custom animations.

discord: dillonpw

#### Related issues & labels (optional)

- Closes Improve Custom Animations code example for View Transitions guide #8787
- Suggested label: [improve documentation](https://github.com/withastro/docs/labels/improve%20documentation)

### CODE REVIEW

**Comment by sarah11918** [comment]
Thank you, @Dillonpw ! Will ask @martrapp for a technical review on this one!

**Comment by martrapp** [comment]
Sure thing! Will do tomorrow.

**Review by martrapp — CHANGES_REQUESTED** [review]
A very cool and helpful example, @Dillonpw! 

My suggestions are mainly aimed at making the code blocks as similar as everywhere else in the docs: titles, single quotation marks, concentration on the essentials and omitting what is not directly related to the topic: How to make good layout components, for example, is already described elsewhere ;-)

It is important to me that the out-animation really takes out the old image and does not push it in.

And yes, even if 'crash' was meant to be fun, it does evoke rather unpleasant associations during software development ;-)  

In terms of content, I think these changes are a real enrichment. The wording in the body text should be approved by native speakers.

File: src/content/docs/en/guides/view-transitions.mdx [hunk 1]
```diff
@@ -221,34 +221,77 @@ export interface TransitionDirectionalAnimations {
...
 
 ```astro
 ---
 const anim = {
   old: {
-    name: 'fadeIn',
-    duration: '0.2s',
-    easing: 'linear',
-    fillMode: 'forwards',
+    name: "crash",
```
Comment (martrapp) [inline src/content/docs/en/guides/view-transitions.mdx:272]: Please use single quotation marks in the frontmatter, as is usually done in the documents

File: src/content/docs/en/guides/view-transitions.mdx [hunk 2]
```diff
@@ -221,34 +221,77 @@ export interface TransitionDirectionalAnimations {
 }
 ```
 
-The following example shows all the necessary properties to define a custom `fade` animation:
+The following example shows all the necessary properties to define a custom transition animation.
+
+In this instance, the custom animation `crash` is defined inside a `<style is:global>` tag in your root layout file:
```
Comment (martrapp) [inline src/content/docs/en/guides/view-transitions.mdx:226]: "crash" sounds so destructive and somehow like "broken". Surely there's a more cheerful name?

File: src/content/docs/en/guides/view-transitions.mdx [hunk 3]
```diff
@@ -221,34 +221,77 @@ export interface TransitionDirectionalAnimations {
...
 const anim = {
   old: {
-    name: 'fadeIn',
-    duration: '0.2s',
-    easing: 'linear',
-    fillMode: 'forwards',
+    name: "crash",
+    duration: "0.5s",
+    easing: "ease-in",
+    fillMode: "forwards",
```
Comment (martrapp) [inline src/content/docs/en/guides/view-transitions.mdx:275]:
```suggestion
    direction: "reverse",
```

File: src/content/docs/en/guides/view-transitions.mdx [hunk 4]
```diff
@@ -221,34 +221,77 @@ export interface TransitionDirectionalAnimations {
...
   new: {
-    name: 'fadeOut',
-    duration: '0.3s',
-    easing: 'linear',
-    fillMode: 'backwards',
-  }
+    name: "crash",
+    duration: "0.5s",
+    easing: "ease-in-out",
+    fillMode: "backwards",
```
Comment (martrapp) [inline src/content/docs/en/guides/view-transitions.mdx:281]:
```suggestion
```

File: src/content/docs/en/guides/view-transitions.mdx [hunk 5]
```diff
@@ -221,34 +221,77 @@ export interface TransitionDirectionalAnimations {
...
+  forwards: anim,
+  backwards: anim,
 };
 ---
 
-<header transition:animate={myFade}> ... </header>
+<header transition:animate={customTransition}> ... </header>
 ```
 
+
```
Comment (martrapp) [inline src/content/docs/en/guides/view-transitions.mdx:294]: Maybe we should add a sentence saying that you can of course use different objects for forward and backward and different keframe names for old and new?

**Review by martrapp — APPROVED** [review]
Thank you very much, @Dillonpw! Looks good to me!

**Comment by sarah11918** [comment]
Thanks for the review, Martin! I'll do my docs review this week!

File: src/content/docs/en/guides/view-transitions.mdx [hunk 6]
```diff
@@ -221,33 +221,71 @@ export interface TransitionDirectionalAnimations {
 }
 ```
 
-The following example shows all the necessary properties to define a custom `fade` animation:
+The following example shows all the necessary properties to define a custom transition animation.
 
-```astro
+In this instance, the custom animation `bump` is defined inside a `<style is:global>` tag in your root layout file:
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:226]:
```suggestion
The following example shows all the necessary properties to define a custom `bump` animation inside a `<style is:global>` tag in your root layout file:
```
Did we really need two sentences for this? Will this do?  :smile:

File: src/content/docs/en/guides/view-transitions.mdx [hunk 7]
```diff
@@ -221,33 +221,71 @@ export interface TransitionDirectionalAnimations {
 }
 ```
 
-The following example shows all the necessary properties to define a custom `fade` animation:
+The following example shows all the necessary properties to define a custom transition animation.
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:224]:
```suggestion
```

File: src/content/docs/en/guides/view-transitions.mdx [hunk 8]
```diff
@@ -221,33 +221,71 @@ export interface TransitionDirectionalAnimations {
...
+    100% {
+      opacity: 1;
+      transform: scale(1) translateX(0);
+    }
+  }
+</style>
+
+```
+
+Remember to define the animation behavior in the frontmatter of all the components using the animation:
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:262]:
```suggestion
The animation behavior's must be defined in the frontmatter of every component using the animation:
```

File: src/content/docs/en/guides/view-transitions.mdx [hunk 9]
```diff
@@ -221,33 +221,71 @@ export interface TransitionDirectionalAnimations {
...
+const customTransition = {
+  forwards: anim,
+  backwards: anim,
 };
 ---
 
-<header transition:animate={myFade}> ... </header>
+<header transition:animate={customTransition}> ... </header>
 ```
+When customizing animations, take some time to experiment with different variations. You can use different objects for forward and backward, and your keyframe animations for old and new can be different as well.
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:288]:
```suggestion

You have great flexibility when defining custom animations. To achieve your desired result, you may wish to consider unusual combinations such as using different objects for forward and backward, or providing separate keyframe animations for old and new.
```

Is this sort of like what you were going for? It's not really docs' place to say "take some time to experiment", but I did try to make it sound like "you can probably do more than you think -- this feature isn't limiting"

**Review by sarah11918 — COMMENTED** [review]
Thank you for this @Dillonpw , and thank you Martin for the helpful review comments to create this helpful example!

I've left a few tiny suggestions, see what you think!

**Comment by Dillonpw** [comment]
@sarah11918 @martrapp Thank you both for your suggestions, guidance and your time with moving this PR forward!

File: src/content/docs/en/guides/view-transitions.mdx [hunk 10]
```diff
@@ -221,34 +221,72 @@ export interface TransitionDirectionalAnimations {
...
+    100% {
+      opacity: 1;
+      transform: scale(1) translateX(0);
+    }
+  }
+</style>
+
+```
+
+The animation behavior's must be defined in the frontmatter of every component using the animation:
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:261]:
```suggestion
The animation's behavior must be defined in the frontmatter of every component using the animation:
```

**Review by sarah11918 — APPROVED** [review]
Looks good to me, @Dillonpw ! Thanks again, and welcome to Team Docs! :partying_face:

**Review by liruifengv — APPROVED** [review]
LGTM! 🎉

**Pull request merged**